lnwallet/channel.go

```diff
@@ -6498,6 +6498,16 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView,
 		return 0, 0
 	}
 
+	// If after evaluating this prospective view, we find out that we
```

Comment:
Is this going to work well everywhere `availableCommitmentBalance` is used (indirectly)?

I was looking at `LightningChannel.validateFeeRate` and wondering if it would needlessly block fee updates?

Perhaps an alternative is to validate the flow control limits in `channelLink.canSendHtlc`, so that the definition of balance remains unchanged.

Reply:
That's a good observation!

I think `validateFeeRate` ideally shouldn't call this method, as this method really is meant to return the balance available for HTLCs. Instead we instead to a fill view validation with the new feerate.